### DISCUSSION TITLE
Unable to run Flutter integration tests

### DISCUSSION THREAD

**Discussion by fgatti675** [codemagic-ci-cd/codemagic-docs#2430] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hi, I have a flutter app with a single integration test, which I fail to run, after following the instructions.

In my pretest script I run this command to launch the emulator and get dependecies
```
#!/usr/bin/env bash

flutter emulators --launch apple_ios_simulator
flutter pub get
```

Then the test step runs this command:
```
flutter test --machine integration_test --flavor dev
```

which timeouts after 12 min (it takes about one minute locally)

```

> flutter test --machine integration_test --flavor dev
{"protocolVersion":"0.1.1","runnerVersion":"1.24.9","pid":71060,"type":"start","time":0}
{"suite":{"id":0,"platform":"vm","path":"/Users/builder/clone/integration_test/registration_test.dart"},"type":"suite","time":0}
{"test":{"id":1,"name":"loading /Users/builder/clone/integration_test/registration_test.dart","suiteID":0,"groupIDs":[],"metadata":{"skip":false,"skipReason":null},"line":null,"column":null,"url":null},"type":"testStart","time":1}
{"count":1,"time":7,"type":"allSuites"}
Running pod install...                                             28.5s
Running Xcode build...                                          {"testID":1,"error":"TimeoutException after 0:12:00.000000: Test timed out after 12 minutes.","stackTrace":"package:test_api/src/backend/invoker.dart 338:28  Invoker._handleError.<fn>\ndart:async/zone.dart 1391:47                      _rootRun\ndart:async/zone.dart 1301:19                      _CustomZone.run\npackage:test_api/src/backend/invoker.dart 336:10  Invoker._handleError\npackage:test_api/src/backend/invoker.dart 291:9   Invoker.heartbeat.<fn>.<fn>\ndart:async/zone.dart 1399:13                      _rootRun\ndart:async/zone.dart 1301:19                      _CustomZone.run\npackage:test_api/src/backend/invoker.dart 290:38  Invoker.heartbeat.<fn>\ndart:async-patch/timer_patch.dart 18:15           Timer._createTimer.<fn>\ndart:isolate-patch/timer_impl.dart 398:19         _Timer._runTimers\ndart:isolate-patch/timer_impl.dart 429:5          _Timer._handleMessage\ndart:isolate-patch/isolate_patch.dart 184:12      _RawReceivePort._handleMessage\n","isFailure":false,"type":"error","time":720041}
{"testID":1,"result":"error","skipped":false,"hidden":false,"type":"testDone","time":720047}
{"success":false,"type":"done","time":720050}
```

I have tried a bunch of alternatives like building the app in the previous step to reduce the times, but no luck.
Also I haven't found a way to change that timeout.
Any help is greatly appreciated

### Expected results

The test passes and the build continues

### Actual results

Integration test timeouts after 12 min

### Build id (optional)

65eb4d5aad78534468174661

**Comment by himesh-cm** [comment, marked as answer]
Hi @fgatti675 

The actual error
​"error":"TimeoutException after 0:12:00.000000: Test timed out after 12 minutes.
​
This is a known issue with Flutter, which can be c intermittently. Please check this [GH](https://github.com/flutter/flutter/issues/105913) issue for more information.
The workaround is to use the flutter drive command.
​
Let me know if this answers your query.

**Comment by himesh-cm** [comment]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
